torch/csrc/jit/runtime/symbolic_shape_registry.cpp

```diff
@@ -629,6 +642,85 @@ std::unordered_map<const FunctionSchema*, std::shared_ptr<Graph>>
 // CompilationUnit that holds all these Functions and keeps them alive.
 auto compilation_unit = std::make_shared<CompilationUnit>();
 
+const at::optional<const FunctionSchema*> getInplaceVariant(
+    const FunctionSchema& base_schema) {
+  auto& inplace_variants =
+      getAllOperatorsFor(c10::Symbol::fromQualString(base_schema.name() + "_"));
+
+  for (const auto& variant : inplace_variants) {
+    // Need to check that all args are the same except for the first, which
+    // is almost the same except for the Alias info
+    const FunctionSchema* schema = &variant->schema();
+    // TODO: try this out. If the tests fail, then we know something is wrong
+    if (!schema->isSubtypeOf(base_schema, true)) {
+      continue;
+    }
+
+    Argument self_arg = schema->arguments()[0];
+    if (*self_arg.type() != *base_schema.arguments()[0].type()) {
+      continue;
+    }
+    if (!self_arg.alias_info()->isWrite()) {
+      continue;
+    }
+    // Lastly check the return types are the same
+    if (schema->returns().size() != 1) {
+      continue;
+    }
+    Argument ret_arg = schema->returns()[0];
+    if (!ret_arg.alias_info()->isWrite()) {
+      continue;
+    }
```

Comment:
ugh, i dont know if we really need to check this, it should be pretty alrready covered by firrst check but maybe it's finee